components/org.wso2.carbon.identity.data.publisher.audit.common/src/main/java/org/wso2/carbon/identity/data/publisher/audit/common/AuditDataPublisherUtils.java

```diff
@@ -22,30 +22,42 @@
 import org.wso2.carbon.base.MultitenantConstants;
 import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
 
+/**
+ * Audit data publisher related utilities.
+ */
 public class AuditDataPublisherUtils {
 
-    public static String[] getTenantDomains(String spTenantDomain, String userTenantDomain) {
+    private AuditDataPublisherUtils() {     // Prevent initializing
+    }
 
-        if (StringUtils.isBlank(userTenantDomain) || userTenantDomain.equalsIgnoreCase(AuditDataPublisherConstants
-                .NOT_AVAILABLE)) {
-            return new String[]{spTenantDomain};
-        }
-        if (StringUtils.isBlank(spTenantDomain) || userTenantDomain.equalsIgnoreCase(AuditDataPublisherConstants
-                .NOT_AVAILABLE)) {
-            return new String[]{userTenantDomain};
-        }
-        if (spTenantDomain.equalsIgnoreCase(userTenantDomain)) {
-            return new String[]{userTenantDomain};
+    /**
+     * Get the tenant domains to publish to based on action holder tenant domain and user tenant domain.
+     *
+     * @param actionHolderTenantDomain Action holder tenant domain
+     * @param userTenantDomain         User tenant domain
+     * @return The list of tenant domains to publish to
+     */
+    public static String[] getTenantDomains(String actionHolderTenantDomain, String userTenantDomain) {
+        String[] tenantDomains;
+        if (StringUtils.isBlank(userTenantDomain) ||
+                userTenantDomain.equalsIgnoreCase(AuditDataPublisherConstants.NOT_AVAILABLE)) {
+            tenantDomains = new String[]{actionHolderTenantDomain};
+        } else if (StringUtils.isBlank(actionHolderTenantDomain) ||
+                userTenantDomain.equalsIgnoreCase(AuditDataPublisherConstants.NOT_AVAILABLE)) {
```

Comment:
Is this the correct condition to be checked ?

Reply:
I changed the implementation to a more readable logic.